pkg/sink/postgres.go

```diff
@@ -367,26 +344,28 @@ parse:
 }
 
 func (p *PGXSink) performDDL(count int, ddl string) (err error) {
-	ctx := context.Background()
 	if count == 1 {
 		// if the ddl contains only one command tag, than do not run it into begin/commit block
 		// because command like "CREATE INDEX CONCURRENTLY" can't be run inside transaction
-		if _, err = p.conn.Exec(ctx, "commit"); err != nil {
+		if err = p.endPipeline(); err != nil {
 			return err
 		}
+		p.startPipeline()
 	}
-	if _, err = p.conn.Exec(ctx, ddl); err == nil && count == 1 {
-		if _, err = p.conn.Exec(ctx, "begin"); err != nil {
-			return err
-		}
+
+	if err = p.endPipeline(); err != nil {
```

Comment:
It seems that we don't always end the transaction here in the previous implementation. Why do we always terminate the pipeline here?

Reply:
There are two reasons here:
1. Because DDL operations must be executed first to ensure the success of subsequent schema refresh operations. 
2. The DDL operations may contain multiple statement which does not allowed in prepare statement.

Therefore, consider that DDL operations are not very frequent. I choose terminate pipeline here.

Reply:
After revisiting the `parseDDL`, I noticed that we couldn't simply restart the pipeline because by doing that, the preceding DMLs in the same transaction will be committed partially.

We will have to first remove the preceding DMLs of the same transaction and then end the pipeline, and finally, we probably need to apply the transaction that contains the DDL in the old way.

Reply:
it looks like we can remove pendingChanges and execute DDL operation which will also contains DML operation.